core/domain/suggestion_jobs_one_off.py

```diff
@@ -348,3 +348,45 @@ def reduce(key, values):
             yield (key, len(values))
         else:
             yield (key, values)
+
+
+class ContentSuggestionFormatUpdateJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that migrates the format of content suggestions."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        if item.suggestion_type != feconf.SUGGESTION_TYPE_EDIT_STATE_CONTENT:
+            return
+
+        changes_made = False
+        if 'old_value' in item.change_cmd:
+            del item.change_cmd['old_value']
+            changes_made = True
+            yield ('CHANGED - Removed old_value', item.id)
+        if 'content_id' in item.change_cmd['new_value']:
+            del item.change_cmd['new_value']['content_id']
+            changes_made = True
+            yield ('CHANGED - Removed content_id', item.id)
+
+        # Validate the eventual suggestion format.
+        try:
+            assert set(item.change_cmd.keys()) == set([
+                'state_name', 'cmd', 'new_value', 'property_name'])
```

Comment:
Do we need some changes in service/domain layer of suggestion to support this new format of command?

Reply:
![image](https://user-images.githubusercontent.com/16653571/109922808-f123ae80-7ce3-11eb-9dca-8db6a3e5b232.png)
I took a quick pass on the apply_changelist function and found we create subtitleHtml from the new_value filed in the change_cmd and if we are going to remove the content_id, then we won't be able to create the subtitleHtml, right? (not sure!) Can you please confirm and if that's true then it worth checking why the backend test for the one-off-job didn't fail?

Reply:
Also, how these old_values keys were added in the suggestion in the first place, do we need to change the functionality such that it doesn't add such value in the future (And also add validation check?)?

Reply:
See get_change_list_for_accepting_suggestion() in suggestion_registry.py. It creates a Change object with the data and adds those two fields manually.

This is presumably because it made no sense to use the actual fields that came with the suggestion, since those fields might be out of date already by the time the suggestion is accepted. However, whoever did this change seems to have totally forgotten about running a prod job to migrate old data, which is what this is fixing. It seems reasonable that they are in the datastore still because they haven't been changed since that modification was made.

Reply:
Also, just to confirm, I did test (manually) adding a completely new suggestion, and the current format is the correct one. Most of the current suggestions in prod look like that, but there are a minority of older ones which don't and this job fixes that.

Reply:
Through offline discussion we found:

1. API is still allowing and saving (in datastore) unwanted fields in change_cmd of suggestion.
2. We are going to do this: (a) Get this PR in as-is, (b) Tomorrow @seanlip will make a PR that disables the "add content suggestion" functionality in the learner view and get that in for the release, too. 

[Note: we are going to drop the "suggest" functionality altogether from the learner's view. We have been working through the suggestions for the curated lessons and they are not helpful at all.]